lib/manageiq/appliance_console/database_admin.rb

```diff
@@ -0,0 +1,119 @@
+require 'manageiq/appliance_console/errors'
+
+module ManageIQ
+  module ApplianceConsole
+    class DatabaseAdmin < HighLine
```

Comment:
Why is this needed? I don't think we do this anywhere else.

Reply:
_@NickLaMuro realizes he is going to be writing another_ :book: _to explain this... \*sigh\*_

tl; dr  #Reasons™

To start off:

Actually [we do](https://github.com/ManageIQ/manageiq-appliance_console/blob/3f49d2de/spec/prompts_spec.rb#L22), it is just not done within the codebase normally when we run `bin/appliance_console`.

In `bin/appliance_console` in "production", by contrast, [import](https://github.com/ManageIQ/manageiq-appliance_console/blob/3f49d2de/bin/appliance_console#L14-L15), [include](https://github.com/ManageIQ/manageiq-appliance_console/blob/3f49d2de/bin/appliance_console#L21), and even [configure](https://github.com/ManageIQ/manageiq-appliance_console/blob/3f49d2de/bin/appliance_console#L35-L36) `HighLine` within the `bin/appliance_console` to make the DSL methods "globally available" via `$terminal` [[ref]](https://github.com/JEG2/highline/blob/42f1e7c3/lib/highline/import.rb#L11).

For various reasons, #thisSucks™ outside of a quick single file script because it means we can't test the configuration we set there, and I don't have access to that in the tests unless I include `highline/import` within the tests, which would probably mess with the `prompts_spec.rb` that I wasn't willing to deal with.

We have also basically been punting on the `HighLine` DSL in the specs as well, so anything that actually used the DSL in it's class (`certificate_authority.rb` for example) was either stubbing it or just not running those methods.

* * *

Now doing this has it's drawbacks.  Technically speaking, we are using a different instance of `HighLine` here that isn't `$terminal`, so the `$terminal.wrap_at=` and `$terminal.page_at=` calls are not respected in this case.  My plan was to get out of using the global state for this with the `UI.run` stuff, so for now we are "split brained" with the addition of this which might not be the best.  There might also be some weirdness with in practice when there is both the `$terminal` DSL and the inherited one exist at the same time, but Ruby should just favor the local methods over the ones that are forwarded on from `Kernel`.

That said, I don't think there is much that takes advantage `page_at` or `wrap_at` of this in `DatabaseAdmin`, so I think we are fine, for now (any of the major printing we might see probably will be probably from `AwesomeSpawn`, so that probably won't be managed by `HighLine`).  But moving forward, this is part of the reason I wanted to split components the parts that need the `HighLine` DSL ("UI" specific) from the parts that are doing the `#activate` code and would be shared with the CLI (will add this to my case in the "GH issue I have yet to create").

Reply:
But given that every I/O method we use here with the exception of `#say` is actually defined on `Prompts` would it be better to have that class inherit from `Highline` then get the methods from the include?

I feel like it's easier to understand when there is only one place that the I/O methods are coming from.

Reply:
@carbonin well, `Prompts` is a `module`... so not sure that would work... 😞

Reply:
> I feel like it's easier to understand when there is only one place that the I/O methods are coming from.

But yes, 100% agree with you on this, just not sure how much of this I should try and tackle here.  Open to suggestions.

Reply:
Ah right, damn.